packages/sw-precaching/src/lib/controllers/base-manager.js

```diff
@@ -0,0 +1,171 @@
+import ErrorFactory from '../error-factory';
+import StringPrecacheEntry from
+  '../models/precache-entries/string-precache-entry.js';
+import ObjectPrecacheEntry from
+  '../models/precache-entries/object-precache-entry.js';
+
+class BaseCacheManager {
+  constructor() {
+    this._entriesToCache = {};
+  }
+
+  cache(rawEntries) {
+    rawEntries.forEach((rawEntry) => {
+      const precacheEntry = this._parseEntry(rawEntry);
+      this._addEntryToInstallList(precacheEntry);
+    });
```

Comment:
Is there any reason to not directly pass `this._parseEntry(rawEntry)` to `_addEntryToInstallList` here?